src/monodroid/jni/xamarin-android-app-context.cc

```diff
@@ -5,12 +5,12 @@
 
 using namespace xamarin::android::internal;
 
-void* MonodroidRuntime::get_function_pointer (uint32_t mono_image_index, uint32_t class_token, uint32_t method_token) noexcept
+void MonodroidRuntime::get_function_pointer (uint32_t mono_image_index, uint32_t class_index, uint32_t method_token, void *&target_ptr) noexcept
```

Comment:
Why rename `class_token` to `class_index`?  `mono_class_get()` uses `type_token` as the parameter name, so a `_token` suffix appears to make more sense?

Reply:
Because token is no longer used to look up a class. Each generated method passes an index into cache array of class instances:

```
; marshal_methods_class_cache
@marshal_methods_class_cache = global [104 x %struct.MarshalMethodsManagedClass] [
        ; 0
        %struct.MarshalMethodsManagedClass {
                i32 33554780, ; token 0x200015c; class name: Android.Views.View, Mono.Android, Version=0.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065
                %struct.MonoClass* null; klass
        }, 
        ; 1
        %struct.MarshalMethodsManagedClass {
                i32 33554448, ; token 0x2000010; class name: AndroidX.Activity.ContextAware.IOnContextAvailableListenerInvoker, Xamarin.AndroidX.Activity, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
                %struct.MonoClass* null; klass
        }, 
        ; 2
        %struct.MarshalMethodsManagedClass {
                i32 33555145, ; token 0x20002c9; class name: Android.Content.IDialogInterfaceOnClickListenerInvoker, Mono.Android, Version=0.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065
                %struct.MonoClass* null; klass
        }, 
        ; 3
```

It's faster to look up this way:

```c++
MarshalMethodsManagedClass &klass = marshal_methods_class_cache[class_index];
if (klass.klass == nullptr) {
  klass.klass = mono_class_get (image, klass.token);
}
```